mono/metadata/marshal.c

```diff
@@ -8799,6 +8819,7 @@ mono_marshal_emit_managed_wrapper (MonoMethodBuilder *mb, MonoMethodSignature *i
 		/* mono_threads_detach_coop (); */
 		mono_mb_emit_ldloc (mb, attach_cookie_local);
 		mono_mb_emit_ldloc_addr (mb, attach_dummy_local);
+		mono_mb_emit_ldloc (mb, enter_blocking_local);
```

Comment:
instead of passing this `enter_blocking` variable to `mono_threads_detach_coop`, you should just check if it's equal to `1` or `0` and call `mono_threads_detach_coop` accordingly. If it's `1`, it means we rethrew the exception so we don't even want to detach, and if it's `0` it mean we need to detach.